src/indigenousLayers.js

```diff
@@ -38,49 +37,35 @@ L.LayerGroup.IndigenousLandsTerritoriesLayer = L.LayerGroup.extend(
                 (function() {
                     var zoom = self._map.getZoom(), origin = self._map.getCenter() ;
                     var $ = window.jQuery;
+                    var ILL_url;
+
+                    if(this.name == "Territories"){
+                        ILL_url = "https://native-land.ca/api/index.php?maps=territories&position=" + parseInt(origin.lat) + "," + parseInt(origin.lng);
+                    }
+                    else if(this.name == "Languages"){
+                        ILL_url = "https://native-land.ca/api/index.php?maps=languages&position=" + parseInt(origin.lat) + "," + parseInt(origin.lng);
+                    }
+                    else{
+                        ILL_url = "https://native-land.ca/api/index.php?maps=treaties&position=" + parseInt(origin.lat) + "," + parseInt(origin.lng);
+                    }
```

Comment:
Hi @sagarpreet-chadha , I tried both passing the ILL_url as a parameter and this if else block like this. But still getting the same cors error. I haven't been able to figure out another way yet.

Reply:
Here as we are defining URL again , so lets pass `name` variable everywhere in arguments,etc. and remove `URL` variable in function argument . I hope it makes sense .